### PR TITLE
Bugfix/scatter

### PR DESCRIPTION
**Pull request by Alex--wu**
1. Fix the bug #1164 
2. Add new feature for the lineChart: add distribution lines like how the scatterChart showing.

### CODE REVIEW

File: src/models/lineChart.js [hunk 1]
```diff
@@ -306,6 +351,30 @@ nv.models.lineChart = function() {
...
+                tooltip.hidden(true);
+                if(showDistX){
+                    container.select('.nv-distributionX .nv-series-' + evt.seriesIndex + ' .nv-distx-' + evt.pointIndex)
+                        .attr('y1', 0);
+                }
+                if(showDistY){
+                    container.select('.nv-distributionY .nv-series-' + evt.seriesIndex + ' .nv-disty-' + evt.pointIndex)
+                        .attr('x2', distY.size());
+                }
+            });
```
Comment (liquidpele) [inline src/models/lineChart.js]: Why are you moving these functions here?   They'll get overwritten every chart.update() in this case.

**Comment by liquidpele** [comment]
Hmm, well the unit tests failed, and it looks like you're trying to add in a feature with the bug fix... please submit 2 different pull requests, it's hard to judge code with stuff all put into a single pull request like that.  For instance, the bug fix looks good but I'm not sure I even like the idea of the distribution lines on the line chart... I'm not sure why I left them on the scatter chart honestly :p

File: src/models/lineChart.js (hunk 1 above)
Comment (Alex--wu) [inline src/models/lineChart.js]: I just make it like the scatterChart. You are right , I will move it
outside the chart scope. Also I will move the one in scatterChart outside
the scope too. And I will do some optimization on the scatterChart.

2015-07-27 19:18 GMT+02:00 liquidpele <notifications@github.com>:

Comment (Alex--wu) [inline src/models/lineChart.js]: I will do that tomorrow. Too busy these days.

2015-07-27 19:30 GMT+02:00 Shengyuan Wu <shengyuan.w@gmail.com>:

**Pull request closed without merging**